public/app/plugins/panel/heatmap-new/ColorScale.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState, useEffect } from 'react';
+import { css } from '@emotion/css';
+import { GrafanaTheme2 } from '@grafana/data';
+import { useTheme2 } from '@grafana/ui';
+
+type Props = {
+  colorPalette: string[];
+};
+
+export const ColorScale = ({ colorPalette }: Props) => {
+  const [colors, setColors] = useState<string[]>([]);
+  useEffect(() => {
+    setColors(getGradientStops({ colorArray: colorPalette }));
+  }, [colorPalette]);
+
+  const theme = useTheme2();
+  const styles = getStyles(theme, colors);
+
+  // @ts-ignore
+  return (
+    <div className={styles.scaleWrapper}>
+      <div className={styles.scaleGradient} />
+    </div>
+  );
+};
+
+const getGradientStops = ({ colorArray, stops = 10 }: { colorArray: string[]; stops?: number }): string[] => {
+  const gradientEnd = colorArray[colorArray.length - 1];
+  const colorCount = colorArray.length;
```

Comment:
maybe swap these two and reuse `colorCount` var?

```suggestion
  const colorCount = colorArray.length;
  const gradientEnd = colorArray[colorCount - 1];
```